components/brave_ads/core/internal/prefs/pref_registry.cc

```diff
@@ -25,6 +25,8 @@ void RegisterProfilePrefs(user_prefs::PrefRegistrySyncable* registry) {
   registry->RegisterBooleanPref(prefs::kNotificationAdDidFallbackToCustom,
                                 false);
 
+  registry->RegisterBooleanPref(prefs::kOptedInToSearchResultAds, true);
```

Comment:
Sanity checked with @aseren and this logic is correct for desktop, Android and iOS Thanks